src/searching/binary_search.rs

```diff
@@ -0,0 +1,220 @@
+use std::cmp::Ordering;
+/// ## Binary Search algorithm.
+/// Iterating over the array by dividing half of the array and returns the index value of the target element in the array.
```

Comment:
Maybe we can improve upon this by being more concrete about how the algorithm works and then giving implementation details.

Imagine that you have never seen this algorithm before, how would you explain it? Then imagine that you have only ever seen the recursive implementation, how would you explain this one?